pages/status/basic-error-catch.vue

```diff
@@ -0,0 +1,30 @@
+<template>
+  <div style="top: 111px; position: absolute">
+    <ComponentToTest></ComponentToTest>
+  </div>
+</template>
+
+<script>
+// import ComponentToTest from "@/components/status/StatusBar.vue";
+import { StatusBar as ComponentToTest } from "../../dist/mantarray.common";
+
+import { STATUS } from "@/store/modules/flask/enums";
+
+export default {
+  components: {
+    ComponentToTest,
+  },
+  created: function () {
+    this.$store.commit(
+      "flask/set_status_uuid",
+      STATUS.MESSAGE.CALIBRATION_NEEDED
+    );
+    setTimeout(this.reset_error, 10);
```

Comment:
please remove the timeout as we discussed Friday morning and this morning.  Please instead have some dummy button off to the side or a keypress event within this test page that triggers a change in Vuex State.  Then Testcafe can click on the dummy button or press the key.

Please have an E2E test of the full path of screenshots for: Status is not an error (e.g. calibration needed), status changes to error and pops up the dialog, clicking on the Okay button closes the pop-up and the status widget changes to say "Shutting down"

Reply:
I  tried using a button on the page of `basic-error-catch.vue`  after the yestarday's discussion, with the event of a `onclick` mutating the store to `ERROR` .   The `this.$bvModal.show("error-catch");`  is executed but the `ErrorCatchWidget` doesn't appear on the page, so had to use the present method.

For now the present verification has no timing issue or errors, so prefer to have the same.  

The Second part of E2E to capture the "Shutting down", I feel is covered already in the Jest testing.

Reply:
The visual appearance and disappearance of the pop-up is important to check.  The "shutting down" behavior in Jest is triggered by directly invoking a method (`remove_errorcatch`)---directly invoking a method is not something that ever happens in the real application, so we need to confirm that clicking the Okay button actually causes the pop-up to disappear and the message to change.

At some point hopefully we can find a more robust way to test emitted events instead of directly invoking methods in Jest, but I haven't figured a way out yet.

If you are unable to figure out a way to E2E test this, then we can try together during our morning meeting